scripts/retrieve_training_metrics.py

```diff
@@ -3,20 +3,108 @@
 import argparse
 import logging
 import sys
-from os.path import abspath
 
 import requests
+import taskcluster
+
+from bugbug.utils import get_taskcluster_options
 
 LATEST_URI = "train_{}.latest"
 VERSIONED_URI = "train_{}.{}"
 DATED_VERSIONED_URI = "train_{}.{}.{}"
-BASE_URL = "https://index.taskcluster.net/v1/task/project.relman.bugbug.{}/artifacts/public/metrics.json"
+PROJECT_PREFIX = "project.relman.bugbug.{}"
+BASE_URL = "https://index.taskcluster.net/v1/task/{}/artifacts/public/metrics.json"
+NAMESPACE_URI = "project.relman.bugbug.{}"
 
 LOGGER = logging.getLogger(__name__)
 
 logging.basicConfig(level=logging.INFO)
 
 
+def get_task_metrics_from_uri(index_uri):
+    index_url = BASE_URL.format(index_uri)
+    LOGGER.info(f"Retrieving metrics from {index_url}")
+    r = requests.get(index_url)
+
+    if r.status_code == 404:
+        LOGGER.error(f"File not found for URL {index_url}, check your arguments")
+        sys.exit(1)
+
+    r.raise_for_status()
+
+    return r
+
+
+def get_namespaces(index, index_uri):
+    index_namespaces = index.listNamespaces(index_uri)
+
+    return index_namespaces["namespaces"]
+
+
+def is_later_or_equal(partial_date, from_date):
+    for partial_date_part, from_date_part in zip(partial_date, from_date):
+        if int(partial_date_part) > int(from_date_part):
+            return True
+        elif int(partial_date_part) < int(from_date_part):
+            return False
+        else:
+            continue
+
+    return True
+
+
+def get_task_metrics_from_date(model, version, date):
+    options = get_taskcluster_options()
+
+    index = taskcluster.Index(options)
+
+    index.ping()
+
+    # Split the date
+    from_date = date.split(".")
+
+    uris = []
```

Comment:
I wonder if we can't find nicer names for the variables, I've had to read the code a few times to understand it

Reply:
I will try to come with better variable names